### PR TITLE
Support types besides strings in DataSync's filterWhere

### PR DESCRIPTION
**Pull request by mpscholten**

### CODE REVIEW

File: lib/IHP/DataSync/ihp-querybuilder.js [hunk 1]
```diff
@@ -82,4 +82,18 @@ function query(table) {
...
+        return { tag: 'TextValue', contents: value };
+    } else if (typeof value === "number") {
+        return { tag: 'IntValue', contents: value };
+    } else if (typeof value === "boolean") {
+        return { tag: 'BoolValue', contents: value };
+    } else if (value === null || value === undefined) {
+        return { tag: 'NullValue' };
+    }
+
+    throw new Error('Could no transform JS value to DynamicValue');
```
Comment (amitaibu) [inline lib/IHP/DataSync/ihp-querybuilder.js:96]: Maybe indicate which values are valid?

```suggestion
    throw new Error('Could no transform JS value to DynamicValue (Text, Int, Bool)');
```

Comment (amitaibu) [reply lib/IHP/DataSync/ihp-querybuilder.js:96]: btw, do we support Float?

Comment (mpscholten) [reply lib/IHP/DataSync/ihp-querybuilder.js:96]: Good point 👍 will add this.

Floats are not yet implemented

Comment (mpscholten) [reply lib/IHP/DataSync/ihp-querybuilder.js]: Created an issue for Floats here https://github.com/digitallyinduced/ihp/issues/1298

**Pull request merged**